### ISSUE TITLE
A way to send out welcome emails

### ISSUE THREAD

**Issue by ivaigor96** [dynamicweb/DynamicWeb#146]
We have a scenario on many projects when we launch the site the customer requires to send out some welcome emails. The emails need to be configured so that they use a Swift page as template and that we can insert values into them {{Email:User:Username}} etc.

The email contains the link to the login page, where the user can go through the reset password flow (they start with entering their email and then they recieve another email with an actual token to reset password)

We tried to use emails for this

Because emails is made explicitely for newsletters, it requires a value on all users: "NewsletterAllowed" = true and it requires an unsubscribe link.

This means that we can not use it for the above described welcome emails. I have not found a tool in DW10 that would allow us to achieve the welcome emails. The user provider does not allow setting a page as template, it only allows an actual cshtml file, which is not good enough, because we want to have different versions of the welcome email, and we want to be able to configure it without coding.

The unsubscribe link doesn't make sense for emails such as the welcome email, as it is not something users consent to prior. It is currently hard coded into the HTML body in the send provider unless an SQL value is not overwritten manually from the db. 

We also need to specifically chose which users will recieve the invites, this could also mean using segment search (Which email already convinietly supports). The segment search allows us to split users by country, therefore by language and have different language emails for different users.

We don't need to use emails specifically, but we need some tools in DW10 to handle this cenario as it is coming up in many projects that when go live comes, we don't know how to handle this registration process

**Comment by nicped** [comment]
Thank you for an elaborate description!

So basically you need a setting on an email, e.g. "System email" that will disable NewsletterAllowed check and will not insert unsubscribe - or maybe just 2 explicit settings to disable it.

Maybe with an option to set a flag on a user when the email is send to them?

**Comment by ivaigor96** [comment]
> Thank you for an elaborate description!
> 
> So basically you need a setting on an email, e.g. "System email" that will disable NewsletterAllowed check and will not insert unsubscribe - or maybe just 2 explicit settings to disable it.
> 
> Maybe with an option to set a flag on a user when the email is send to them?

Yes, these 2 settings should be enough to achieve the goals we pursue.
Tracking that a mail was already sent to a user is a good idea, we could then schedule the welcome emails for new users joining the platform post go-live, for example when imported from the integration.

I am hopeful and looking forward to seeing these features in new DW10 versions.

**Comment by nicped** [comment]
I have created an epic, 21227, describing this feature for Q4. It has not been decided yet, but I believe this would be a good feature to make onboarding of new users better.

**Comment by ivaigor96** [comment]
I still don't see anything on the setting on an email, e.g. "System email" unless I am missing something. Regarding option to set a flag on a user when the email is send to them - this is possible to achieve with insure unique recepients checkbox

![Image](https://github.com/user-attachments/assets/c3cbe4e3-6859-4b3b-901a-1ca9fa8f4803)

So the only thing we are missing is the ability to ignore the email permission checkbox, unless it is somehow implemented and I am missing something

![Image](https://github.com/user-attachments/assets/1dd436ec-57d5-4dd0-b673-4c74a80aba35)

The email feature is still the only viable solution for sending out welcome emails for go-live users, since they often don't even come from the data integration, they might be from an Excel file or something, so we always need to send out this reset password welcome email to get people registered in the new webshop and start using the solution

**Comment by nicped** [comment]
I have added https://dev.azure.com/dynamicwebsoftware/Dynamicweb/_workitems/edit/24211 that adds an option on the recipient provider to disregard the email allowed flag.
